src/sonic-yang-mgmt/_sonic_yang_ext.py

```diff
@@ -0,0 +1,616 @@
+                    config[pkey][key] = self.revFindYangTypedValue(key, \
+                        entry[key], leafDict)
+
+    return
+
+"""
+Rev xlate from yang container to table in config DB
+"""
+def revXlateContainer(self, model, yang, config, table):
+
+    # Note: right now containers has only LISTs.
+    # IF container has only one list
+    if isinstance(model['list'], dict):
+        modelList = model['list']
+        # Pass matching list from Yang Json
+        self.revXlateList(modelList, yang[modelList['@name']], config, table)
+    else:
+        # TODO: Container[TABLE] contains multiple lists. [Test Pending]
+        # No instance now.
+        for modelList in model['list']:
+            self.revXlateList(modelList, yang[modelList['@name']], config, table)
+
+    return
+
+"""
+rev xlate ConfigDB json to Yang json
+"""
+def revXlateYangtoConfigDB(self, yangJ, cDbJson):
+
+    yangJ = self.xlateJson
+    cDbJson = self.revXlateJson
+
+    # find table in config DB, use name as a KEY
+    for module_top in yangJ.keys():
+        # module _top will be of from module:top
+        for container in yangJ[module_top].keys():
+            #table = container.split(':')[1]
+            table = container
+            #print("revXlate " + table)
+            cmap = self.confDbYangMap[table]
+            cDbJson[table] = dict()
+            #print(key + "--" + subkey)
+            self.revXlateContainer(cmap['container'], yangJ[module_top][container], \
+                cDbJson[table], table)
+
+    return
+
+"""
+Reverse Translate tp config DB
+"""
+def revXlateConfigDB(self, revXlateFile=None):
+
+    yangJ = self.xlateJson
+    cDbJson = self.revXlateJson
+    # xlation is written in self.xlateJson
+    self.revXlateYangtoConfigDB(yangJ, cDbJson)
+
+    if revXlateFile:
+        with open(revXlateFile, 'w') as f:
+            dump(self.revXlateJson, f, indent=4)
+
+    return
+
+"""
+Find a list in YANG Container
+c = container
+l = list name
+return: list if found else None
+"""
+def findYangList(self, container, listName):
+
+    if isinstance(container['list'], dict):
+        clist = container['list']
+        if clist['@name'] == listName:
+            return clist
+
+    elif isinstance(container['list'], list):
+        clist = [l for l in container['list'] if l['@name'] == listName]
+        return clist[0]
+
+    return None
+
+"""
+Find xpath of the PORT Leaf in PORT container/list. Xpath of Leaf is needed,
+because only leaf can have leafrefs depend on them.
+"""
+def findXpathPortLeaf(self, portName):
+
+    try:
+        table = "PORT"
+        xpath = self.findXpathPort(portName)
+        module, topc, container = self.get_module_top_container(table)
+        list = self.findYangList(container, table+"_LIST")
+        xpath = xpath + "/" + list['key']['@value'].split()[0]
+    except Exception as e:
+        print("find xpath of port Leaf failed")
+        raise e
+
+    return xpath
+
+
+"""
+Find xpath of PORT
+"""
```

Comment:
A function only for PORT table?   Do we have a generic function for all tables?

Reply:
I will address this later, Right now, I can not prioritize this. 
Considering that table name will always be provided to find correct YANG module, we may have some wrapper function for each table anyway. Thx.